datadog_lambda/tracing.py

```diff
@@ -332,7 +332,7 @@ def extract_dd_trace_context(event, lambda_context, extractor=None):
             parent_id,
             sampling_priority,
         ) = extract_context_custom_extractor(extractor, event, lambda_context)
-    elif "headers" in event:
+    elif isinstance(event, (list, dict)) and "headers" in event:
```

Comment:
It varies wildly depending on the integration (see [docs](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-concepts.html#gettingstarted-concepts-event)). But this library seems to have been written basically only for a Dict or Set type event - however Lambda allows the type of `event` to be None, binary data, or a few other options (including a list/array), and since we don't have an guarantee, we have to duck-type it.